boards/m4a-wroom/doc.txt

```diff
@@ -0,0 +1,189 @@
+2 x SPI at maximum
+1 x I2C at maximum
+2 x UART
+
+Since all GPIOs have broken out, GPIOs can be used for different purposes
+in different applications. For flexibility, GPIOs can be listed in various
+peripheral configurations. For example, GPIO13 is used in the ADC channel
+definition and the definition of the MOSI signal of SPI_DEV(0).
+
+This is possible because GPIOs are only used for a specific peripheral
+interface when
+
+- the corresponding peripheral module is used, e.g. `periph_i2c`, or
+- a corresponding init function is called z. `adc_init`, `dac_init` and
+  `pwm_init` or
+- The corresponding peripheral interface is used for the first time,
+  e.g. `spi_aqcuire.
+
+That is, the purpose for which a GPIO is used depends on which module
+or function is used first.
+
+For example, if module periph_i2c is not used, the GPIOs listed in I2C
+configuration can be used for the other purposes.
+
+The following table shows the default board configuration, which is sorted
+according to the defined functionality of GPIOs. This configuration can be
+overridden by  esp32_application_specific_configurations
+"application-specific configurations".
+
+<center>
+Function        | GPIOs  | Remarks |Configuration
+:---------------|:-------|:--------|:----------------------------------
+BUTTON0         | GPIO0  | | |
+DS18B20         | GPIO14 | DS18B20 sensor| |
+MOISTURE SENSOR | GPIO12 | Moisture Analog Sensor v1.3 |  |
+ADC             | GPIO0, GPIO2, GPIO4, GPIO12, GPIO13,\n GPIO14, GPIO15, GPIO25, GPIO26, GPIO27,\n GPIO32, GPIO33, GPIO34, GPIO35, GPIO36,\n GPIO39 | | see  esp32_adc_channels "ADC Channels"
+DAC             | GPIO25, GPIO26 | |  esp32_dac_channels "refer"
+PWM_DEV(0)      | GPIO0, GPIO2, GPIO4, GPIO16, GPIO17 | - |  esp32_pwm_channels "DAC Channels"
+PWM_DEV(1)      | GPIO27, GPIO32, GPIO33 | - |  esp32_pwm_channels "PWM Channels"
+I2C_DEV(0):SDA  | GPIO21 | |  esp32_i2c_interfaces "I2C Interfaces"
+I2C_DEV(0):SCL  | GPIO22 | |  esp32_i2c_interfaces "I2C Interfaces"
+SPI_DEV(0):CLK  | GPIO18 | VSPI is used | |
+SPI_DEV(0):MISO | GPIO19 | VSPI is used | |
+SPI_DEV(0):MOSI | GPIO23 | VSPI is used | |
+SPI_DEV(0):CS0  | GPIO5  | VSPI is used | |
+SPI_DEV(1):CLK  | GPIO14 | HSPI is used | |
+SPI_DEV(1):MISO | GPIO12 | HSPI is used | |
+SPI_DEV(1):MOSI | GPIO13 | HSPI is used | |
+SPI_DEV(1):CS0  | GPIO15 | HSPI is used | |
+UART_DEV(0):TxD | GPIO1  | Console (configuration is fixed) | |
+UART_DEV(0):RxD | GPIO3  | Console (configuration is fixed) | |
+UART_DEV(1):TxD | GPIO10 | not available in **qout** and **qio** flash mode | |
+UART_DEV(1):RxD | GPIO9  | not available in **qout** and **qio** flash mode | |
+UART_DEV(2):TxD | GPIO17  | Serial communication |  |
+UART_DEV(2):RxD | GPIO16  | Serial communication |  |
+</center>
+\n
+@note
+- The configuration of ADC channels contains all ESP32 GPIOs that can be
+  used as ADC channels.
+- The configuration of DAC channels contains all ESP32 GPIOs that can be used
```

Comment:
m4a-wroom GPIOs

Reply:
if it's relative to the whole system, the right way is m4a-wroom GPIOs, yes